stl/inc/xmemory

```diff
@@ -1995,6 +1995,19 @@ struct _Alloc_temporary {
     }
 };
 
+template <class _Alloc>
+_NODISCARD constexpr bool _Allocators_equal(const _Alloc& _Lhs, const _Alloc& _Rhs) noexcept {
+    if
+        _CONSTEXPR_IF(allocator_traits<_Alloc>::is_always_equal::value) {
+            (void) _Lhs;

```

Comment:
why do we need to cast these to void here? is it just because the compiler will complain about them being unused without realizing that they are used in the other branch of the constexpr if?

Reply:
Yes, this is (Microsoft-internal) VSO-486357, which is so frequently encountered that we don't mark every occurrence as TRANSITION (although we could, I suppose).

Reply:
I would prefer an STL-wide update like tagging all of them be its own change.

Reply:
I think the whole purpose of `if constexpr` is that the other branches are not processed by the Compiler. Adding scanning for uses of such variables defeats its purpose

Reply:
They still have to be parsed (it’s not “token soup”) and other compilers avoid warning in this scenario.